src/test/java/com/kenticocloud/delivery/DeliveryClientTest.java

```diff
@@ -504,6 +513,90 @@ public void testGetStronglyTypedItemByRegisteringType() throws Exception {
         Assert.assertTrue(itemObj instanceof ArticleItem);
     }
 
+    @Test
+    public void testGetStronglyTypedItemAutomaticallyAddsSystemType() throws Exception {
+        String projectId = "02a70003-e864-464e-b62c-e0ede97deb8c";
+
+        this.serverBootstrap.registerHandler(
+                String.format("/%s/%s", projectId, "items/on_roasts"),
+                (request, response, context) -> {
+                    String uri = String.format("http://testserver%s", request.getRequestLine().getUri());
+
+                    List<NameValuePair> nameValuePairs =
+                            URLEncodedUtils.parse(URI.create(uri), Charset.defaultCharset());
+                    Assert.assertEquals(1, nameValuePairs.stream()
+                            .filter(nameValuePair -> nameValuePair.getName().equals("system.type"))
+                            .count());
+
+                    Map<String, String> params = convertNameValuePairsToMap(nameValuePairs);
+
+                    Assert.assertTrue(params.containsKey("system.type"));
+                    Assert.assertEquals("article", params.get("system.type"));
+
+                    response.setEntity(
+                            new InputStreamEntity(
+                                    this.getClass().getResourceAsStream("SampleContentItem.json")
+                            )
+                    );
+                });
+        HttpHost httpHost = this.start();
+        DeliveryClient client = new DeliveryClient(projectId);
+        client.registerType("article", ArticleItem.class);
+
+        //modify default baseurl to point to test server, this is private so using reflection
+        String testServerUri = httpHost.toURI() + "/%s";
+        Field deliveryOptionsField = client.getClass().getDeclaredField("deliveryOptions");
+        deliveryOptionsField.setAccessible(true);
+        ((DeliveryOptions) deliveryOptionsField.get(client)).setProductionEndpoint(testServerUri);
+
+        ArticleItem itemObj = client.getItem("on_roasts", ArticleItem.class);
+        Assert.assertNotNull(itemObj);
```

Comment:
the tests are not conclusive. they'd pass even if the new type detection logic was not present because you're filtering by codename in the first place.

Reply:
Ultimately, the test hits a mocked server, which limits it's usefulness for the edge you are bringing up, but the mock server is checking for the parameter to be present.  I would expect that in a real world scenario, if you filtered by system.type=article, and codename=some_non_article, that the API wouldn't respond with any results, correct?

Reply:
This test hits the view a content item API, which requires a code name: https://developer.kenticocloud.com/v1/reference#view-a-content-item

Reply:
correct. 👍 i didn't notice the check on the mocked server. merging